### PR TITLE
COST-547 - introduce configurable PVC

### PR DESCRIPTION
**Pull request by maskarb**
- PVC is configurable thru the CR
- If no PVC config is provided, a default is used
- reports directory is initialized in deployment as EmptyDir
- EmptyDir is replaced with PVC
- The Pod reboots after deployment modification, and new Pod access the new PVC

Testing:
1. deploy with the default CR and ensure that the built-in default PVC is created and mounted.
2. keep the PVC from previous step, and redeploy the operator. Create a new CR and ensure that the deployment is mounted onto the previously created PVC (if reports were created in step 1, you could check that they still exist - this test is equivalent to what could happen during an operator upgrade)
3. re-deploy, create a new CR with a defined PVC. Ensure that the specked PVC is created and mounted.
4. create a new PVC. re-deploy. Create a CR with the created PVC spec. Ensure that deployment is mounted onto that created PVC
5. continuing with previous deployment, remove the PVC definition from the CR. Ensure the deployment is re-mounted onto the built-in default PVC. Also check to make sure that the PVC that was unmounted remains within the project

### CODE REVIEW

File: controllers/suite_test.go [hunk 1]
```diff
@@ -241,24 +333,38 @@ func deleteClusterVersion() {
...
-func clusterPrep() {
+func createDeployment(ctx context.Context, deployment *appsv1.Deployment) {
+	Expect(k8sClient.Create(ctx, deployment)).Should(Succeed())
+}
+
+func deleteDeployment(ctx context.Context, deployment *appsv1.Deployment) {
+	Expect(k8sClient.Delete(ctx, deployment)).Should(Succeed())
+}
+
+// func createPVC(ctx context.Context, pvc *corev1.PersistentVolumeClaim) {
```
Comment (boaz0) [inline controllers/suite_test.go:344]: should we keep it or remove this comment?

File: storage/storage.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	dep.Spec.Template.Spec.Volumes[s.vol.index] = *s.vol.volume
+
+	if err := s.Client.Patch(ctx, dep, patch); err != nil {
+		return false, fmt.Errorf("failed to Patch deployment: %v", err)
+	}
+	return true, nil
+}
+
+// ConvertVolume converts the EmptyDir volume in deployment to PVC
+func ConvertVolume(s *Storage, kmCfg *kokumetricscfgv1alpha1.KokuMetricsConfig) (bool, error) {
```
[the remark this answers is not in the document]
Comment (maskarb) [reply storage/storage.go]: Done!

File: controllers/suite_test.go (hunk 1 above)
Comment (maskarb) [reply controllers/suite_test.go]: Yes, I removed these comments.

**Review by boaz0 — APPROVED** [review]

**Review by abaiken — APPROVED** [review]
Great work on this Michael! Went through all of the test scenarios - LGTM 👍

**Pull request merged**